packages/utils/src/binaryUtils.ts

```diff
@@ -0,0 +1,28 @@
+const textEncoder = new TextEncoder()
+const textDecoder = new TextDecoder()
+
+export const binaryToUtf8 = (bytes: Uint8Array): string => {
+    return textDecoder.decode(bytes)
+}
+
+export const utf8ToBinary = (utf8: string): Uint8Array => {
+    return textEncoder.encode(utf8)
+}
+
+export const binaryToHex = (bytes: Uint8Array, addPrefix = false): string => {
+    if (addPrefix) {
+        return `0x${Buffer.from(bytes).toString('hex')}`
+    }
+    return Buffer.from(bytes).toString('hex')
+}
+
+export const hexToBinary = (hex: string): Uint8Array => {
+    if (hex.startsWith('0x')) {
+        hex = hex.slice(2)
+    }
+    return Buffer.from(hex, 'hex')
+}
```

Comment:
Add check to prevent "partial" results due to [data trunctation](https://nodejs.org/api/buffer.html#buffers-and-character-encodings). Return undefined or throw error.

```suggestion
export const hexToBinary = (hex: string): Uint8Array | undefined => {
    if (hex.startsWith('0x')) {
        hex = hex.slice(2)
    }
    const result = Buffer.from(hex, 'hex')
    return hex.length === result.length * 2 ? result : undefined
}
```